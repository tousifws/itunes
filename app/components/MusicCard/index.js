/**
 *
 * Card
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Avatar } from 'antd';
// import { FormattedMessage } from 'react-intl';
import T from '@components/T';

const MusicContent = styled.div`
  display: flex;
  align-items: center;
  height: 120px;
  &:hover {
    cursor: pointer;
  }
`;

function MusicCard({ artworkUrl, title, artistName, collectionName, onClick }) {
  return (
    <Card data-testid="musicCard">
      <MusicContent onClick={onClick}>
        <Avatar src={artworkUrl} shape="square" size={100} />
        <div style={{ paddingLeft: '0.6em' }}>
          <T id="music_title" values={{ title }} />
          <T id="artist_name" values={{ artistName }} />
          <T id="music_collection_name" values={{ collectionName }} />
        </div>
      </MusicContent>
    </Card>
  );
}

MusicCard.propTypes = {
  artworkUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired,
  collectionName: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

export default MusicCard;
