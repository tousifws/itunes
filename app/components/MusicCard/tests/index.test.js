/**
 *
 * Tests for MusicCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import MusicCard from '../index';

describe('<MusicCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<MusicCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 MusicCard component', () => {
    const { getAllByTestId } = renderWithIntl(<MusicCard />);
    expect(getAllByTestId('musicCard').length).toBe(1);
  });
});
