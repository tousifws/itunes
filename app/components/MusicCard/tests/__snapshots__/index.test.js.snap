// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MusicCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered"
      data-testid="musicCard"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="MusicCard__MusicContent-sc-1l6p3fn-0 kfBPgc"
        >
          <span
            class="ant-avatar ant-avatar-square"
            style="width: 100px; height: 100px; line-height: 100px; font-size: 18px;"
          >
            <span
              class="ant-avatar-string"
              style="line-height: 100px; transform: scale(1) translateX(-50%);"
            />
          </span>
          <div
            style="padding-left: 0.6em;"
          >
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Music Title: 
            </p>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Artist: 
            </p>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Music collection name: 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
